examples/sst2/sst2_lib.py

```diff
@@ -0,0 +1,250 @@
+    vocab = collections.OrderedDict()
+    for token in self.special_tokens:
+      vocab[token] = len(vocab)
+
+    # Add all other tokens to the vocab if their frequency is >= min_freq.
+    for token, freq in sorted(
+        # Sort by frequency (from high to low), and then by token string.
+        # This makes sure high frequency tokens get a low token ID.
+        counter.items(),
+            key=lambda token_freq: (-token_freq[1], token_freq[0])):
+      if freq >= min_freq:
+        vocab[token] = len(vocab)
+
+    logging.info('Number of unfiltered tokens: %d', len(counter))
+    logging.info('Vocabulary size: %d', len(vocab))
+    self.vocab = vocab
+
+  def _getitem__(self, key: str):
+    return self.vocab[key]
+
+  def keys(self):
+    return self.vocab.keys()
+
+  def values(self):
+    return self.vocab.values()
+  
+  def __len__(self):
+    return len(self.vocab)
+
+  @property
+  def pad_idx(self):
+    """The index of the padding token."""
+    return self.vocab[self.pad_token]
+
+  @property
+  def unk_idx(self):
+    """The index of the unknown word token."""
+    return self.vocab[self.unk_token]
+
+  @property
+  def bos_idx(self):
+    """The index of the beginning-of-sequence token."""
+    return self.vocab[self.bos_token]
+
+  @property
+  def eos_idx(self):
+    """The index of the end-of-sequence token."""
+    return self.vocab[self.eos_token]
+
+  def load(self, path: str) -> None:
+    """Loads a vocabulary (one token per line) from disk."""
+    vocab = collections.OrderedDict()
+    with open(path, 'rb') as f:
+      for i, token in enumerate(f):
+        assert isinstance(token, bytes), 'Expected byte tokens.'
+        vocab[token.rstrip()] = i
+    logging.info('Loaded vocabulary, size: %d', len(vocab))
+    self.vocab = vocab
+
+  def save(self, path: str) -> None:
+    """Saves the vocabulary to disk."""
+    with open(path, 'wb') as f:
+      for token in self.vocab:
+        assert isinstance(token, bytes), 'Expected byte tokens.'
+        f.write(token + b'\n')
+    logging.info('Saved vocabulary.')
+
+
+def vocab_to_hashtable(vocab: Vocabulary,
+                       unk_idx: int) -> tf.lookup.StaticHashTable:
+  """Returns a TF lookup table (token -> ID) from a vocabulary."""
+  return tf.lookup.StaticHashTable(
+      tf.lookup.KeyValueTensorInitializer(
+          list(vocab.keys()), list(vocab.values())), default_value=unk_idx)
+
+
+def vocab_to_inverse_hashtable(vocab: Vocabulary,
+                               unk_token: bytes) -> tf.lookup.StaticHashTable:
+  """Returns an inverse TF lookup table (ID -> token) from a vocabulary."""
+  return tf.lookup.StaticHashTable(
+      tf.lookup.KeyValueTensorInitializer(
+          list(vocab.values()),
+          list(vocab.keys()),
+          key_dtype=tf.int64,
+          value_dtype=tf.string),
+      default_value=unk_token)
+
+
+def get_bucket_boundaries(bucket_size: int, max_size: int) -> np.ndarray:
```

Comment:
It seems to me all of the functions from here on actually belong in `input_pipeline.py`. Then you could also rename this file to `vocabulary.py`, WDYT?

Reply:
Yes good idea.